src/features/step_definitions/common.steps.js

```diff
@@ -64,6 +74,67 @@ When("I continue without an action", async () => {
   await basePage.continueButton.click();
 });
 
+When("I select {string} and continue", async (option) => {
+
+let choice = "";
+
+  switch (option) {
+    case "conservation covenant": {
+      choice = legalAgreementTypePage.conservationCovenant;
+      break;
+    }
+    case "planning obligation": {
+      choice = legalAgreementTypePage.planningObligation;
+      break;
+    }
+    case "I do not have a legal agreement": {
+      choice = legalAgreementTypePage.doNotHaveDocument;
+      break;
+    }
+  }
+
+  await (await choice).click();
+  await (await legalAgreementTypePage.continueButton).click();
+});
+
+When("I add my fullname or organisation as {string}", async (fullname) => {
+  await legalAgreementAddPartiesPage.legalPartyName.addValue(fullname);
+})
+
+When("I confirm my role as a {string}", async (role) => {
+  //Todo: currently no unique identifiers in code awaiting bug fixes in BNGP-1267
+  await legalAgreementAddPartiesPage.legalPartyRole.waitForExist({ timeout: 5000 });
+  await legalAgreementAddPartiesPage.legalPartyRole.click();
+  await legalAgreementAddPartiesPage.legalContinueButton.click();
+  
+})
+
+When("I enter a valid startdate of {string} for the {string} page", async (date, page) => {
+
+  var arr = date.split('/');
+
+  switch (page) {
+    case "habitat-works-start-date": {
+          await habitatWorksStartDatePage.Day.addValue(arr[0]);
+          await habitatWorksStartDatePage.Month.addValue(arr[1]);
+          await habitatWorksStartDatePage.Year.addValue(arr[2]);
+          await (await habitatWorksStartDatePage.continueButton).click();
```

Comment:
if you have all these selectors in the base page class then you won't have to duplicate this code and you don't have to have the funky `for the X page` part in your step

Reply:
Have updated the selectors that I can control instead